### DISCUSSION TITLE
Purpose & future of Valhalla App

### DISCUSSION THREAD

**Discussion by georg-d** [gis-ops/valhalla-app#43] [General]
@nilsnolde wrote in https://github.com/gis-ops/valhalla-app/issues/33 which is IMHO relevant but not exactly part of the issue

> This repo is not _really_ meant to be a functional routing app, no one is funding this effort. We might take it there along the way, but I can't guarantee anything at this point. Other than being open for PRs and contributions by the community. I'd rather have a proper routing & navigation SDK based on Valhalla that can be integrated into maplibre, iOS/Android, rather than doing these silo projects.

I do not at all see Valhalla App as leading to a silo and an optional component but exactly inverse, so as a door opener for a wide use & adoption, because Valhalla App is not in any way binding the actual routing & navigation part to itself but is _just one_ of many possible GUIs to the engine, _and_ because a good demo'ing GUI is massively lowering the hurdles to play around with the actual engine. For the latter, I see 2 major groups positively influencing Valhalla's adoption:

1. For developers, there are so many possible engines that are alternatives to Valhalla that it somehow has to stand out. Some engines are so well known that devs will have a look at them anyway, e.g. OSRM, and Valhalla is at least not yet one of them. Some engines are for whatever reasons very relevant to the software being developed and thus looked at, e.g. because of specialization like http://www.bbbike.de for bike routing in a very limited area, or because being from the same company (like Microsoft or Meta products will prefer their own routing engine). For the remaining engines, devs will probably ask around in their social network and look at lists like https://wiki.openstreetmap.org/wiki/Routing and because of the sheer amount of options, devs will likely first give those engines a try that are very easy to give a try, and if any of them is suitable, other engines requiring more effort to try won't be looked at – so a Valhalla App that is well usable, provides much functionality and thus aslo delivers routes that match the needs is a door opener, because it allows to quickly get a first impression of the project (quality & alive), functionality and performance. 
2. Users will much more likely use a routing engine with a good GUI that delivers satifying results. The more used an engine is, the more people talk about it & the more visible it will also be for developers (increasing probability it will be used in a software project), and the more the engine's results will be compared with the real world needs – which results in more voiced ideas for improvement, which in turn leads to better and more useful results, which makes the engine more interesting to use in daily life & in a software project.

my 2 cents 🙂

**Comment by nilsnolde** [comment]
Thanks for your thoughts. I can give you my perspective. 

Exactly, this is just a demo GUI for people to play around. In that sense, we'll keep updating it over time to reflect Valhalla dev, but that's pretty much the scope of it and we have no plans of changing that.

> Valhalla is at least not yet one of them

That's a wild claim :smile: ! Without being too cocky, just to give you a bit of the picture: it's the main routing engine for Mapbox since years, runs in Tesla in-car navigation since a similar amount of years, same for BMW since recently, powers lots of routing APIs that you find out there etc. Not many companies say out loud "hey we're really just running Valhalla under the hood", but in terms of industry adoption it's going quite well I think. No offense taken really, but even being as biased as I am, I think it's going good.

IMO we still do have a bit of a problem with open-source (routing) software as a whole, rather on the level of decision-makers not developers. Many people with big cash/budget know and ask others with big cash, Mapbox, ESRI or HERE or so. Rarely someone on that level thinks "out-of-the-box" and goes with a highly specialized niche consultancy offering customizations around existing open-source. From where I'm sitting that's changing fast though, across the board. Once industry adopts even more, it also works out better for an open-source project and its satellite projects, at least with the right-spirited company in the back (i.e. not the companies where Valhalla is currently used by far the most..). Paid feature dev or API pays easily also for great open-source development. There's really not many companies in my field, who are managing to have a (IMO) good balance between open-source and commercial (apart from us haha), and that's Graphhopper and the Pelias/geocode.earth team. They both also serve a niche product with a small team with no plans of growing, and deliver very good value. Which gives them enough money/time to develop a great open-source product. Perfect. And there's still only so much those companies can do for free, neither has a very usable web/mobile app. If the general public would like smth like that they'd have to pay money. And that's not yet really possible for smth as mundane as "just" a routing app, after the previous age of building up a freebie mentality of data-sucking/ad-serving big tech in that area. So not sure we can do much more at this point tbh. Honestly, building what Google Maps mobile app offers on high-level routing stuff with Valhalla (others are not very suitable), would cost more than 1 year for 2-3 core devs, not a small feat with offline capabilities etc, but it'd be of course so much better than Google Maps!:)

**Reply by georg-d** [reply]
> Exactly, this is just a demo GUI for people to play around. In that sense, we'll keep updating it over time to reflect Valhalla dev, but that's pretty much the scope of it and we have no plans of changing that.

Which is IMHO fine. And maybe some folks come around the corner, take that solid base and improve it to match their ideas & needs. 
 
> > Valhalla is at least not yet one of them
> 
> That's a wild claim 😄  
> ...
> Not many companies say out loud "hey we're really just running Valhalla under the hood"

I take that as kind of a proof of my claim 😉 Or at least a major driver into that direction. Compare that to e.g. Google or Bing Maps whose name/logo is visible in nearly every website, app etc. they are embedded into – thus, nearly everybody and all developers know them, while a guy I met last week did never hear of Valhalla despite earning his money with GIS & OSM since a decade. Maybe he has even used Valhalla dozens of times, just without noticing. When such people need a router, Valhalla is not amongst the first 3-4 cadnidates coming to their mind, and in that sense it's not well known enough to be looked at "anyway".

Nonetheless, I am happy to read industry adoption goes well 👍

> IMO we still do have a bit of a problem with open-source (routing) software as a whole, rather on the level of decision-makers not developers. Many people with big cash/budget know and ask others with big cash

+1 just from another domain. 

> And there's still only so much those companies can do for free, neither has a very usable web/mobile app. If the general public would like smth like that they'd have to pay money. And that's not yet really possible for smth as mundane as "just" a routing app

I have the feeling that changes: Organizations like EU or national ministries offer some budget, like e.g. for StreetComplete. Users pay or donate, see Signal messenger or wikipedia, and many small donations can sum up.

>Honestly, building what Google Maps mobile app offers on high-level routing stuff with Valhalla (others are not very suitable), would cost more than 1 year for 2-3 core devs, not a small feat with offline capabilities etc, but it'd be of course so much better than Google Maps!:)

Of course 😉 And I do not think a "full featured Valhalla mobile app" is required – a plugin for e.g. OsmAnd would already suffice and at the same time reduce the development effort because many feature blocks are already existing (map rendering, map usability also when offline, foreground/background activity, management of points,...).
